core/capabilities/remote/executable/parallel_executor_test.go

```diff
@@ -82,6 +84,8 @@ func Test_ExecutingMultipleTasksInParallel(t *testing.T) {
 
 func Test_StopsExecutingMultipleParallelTasksWhenClosed(t *testing.T) {
 	tp := newParallelExecutor(10)
+	err := tp.Start(context.Background())
+	require.NoError(t, err)
```

Comment:
Can we leverage `servicetest.Run` here? It will ensure that `Close` is called on cleanup, even if the test short circuits. It also includes the test context.
```suggestion
	servicetest.Run(t, tp)
```

Reply:
The test explicitly stops the service further down to check behaviour post close.  If I change this to servicetest.Run the test fails as it trys to close the already closed service on test exit.

Reply:
In that case there are a few different ways to solve this. Adding a sub-test would allow using `servicetest.Run` directly, but that could be confusing. Another option is to register the after-close check as a deferred or t.Cleanup func also, but before calling `servicetest.Run`/`tp.Close`, so that it runs afterwards. A third could be a function literal that is called in place, and only serves to scope a defferred call to `tp.Close`, which then allows the final check to stay at the bottom. I suspect the second option would be cleanest (especially if the check can drop the eventually loop? why is that necessary? :thinking:), but the third could be fine too.

Reply:
makes sense, opted for the 2nd option